pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -981,6 +939,48 @@ func (r *ReconcileClusterDeployment) reconcileCompletedProvision(cd *hivev1.Clus
 }
 
 func (r *ReconcileClusterDeployment) clearOutCurrentProvision(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (reconcile.Result, error) {
+	// clear out current provision only if install restart is required
+	if cd.Status.InstallRestarts == 0 && cd.Annotations[tryInstallOnceAnnotation] == "true" {
+		cdLog.Debug("not creating new provision since the deployment is set to try install only once")
+		return reconcile.Result{}, nil
+	}
```

Comment:
@staebler Do we need this code now? We can do the same thing by setting `InstallRestartsLimit` to 0

Reply:
Yes, we still need the code. We cannot stop supporting the annotation, even though there is an easier way to do it now.